wdk-ddi-src/content/fltkernel/ns-fltkernel-_flt_parameters.md

```diff
@@ -105,6 +105,11 @@ The following I/O operations do not have parameters, and therefore do not have a
 - IRP_MJ_SHUTDOWN
 - IRP_MJ_VOLUME_DISMOUNT
 
+For the operations that come in Acquire/Release pairs (ModWrite, CcFlush, Section), there is no guarantee that the release call will happen even after a sucessfull call to the post acquire callback.  An example would be a if an instance detach occurs.

```

Comment:
I suppose that being a Filter Manager owner means I think too hard about the FltMgr implementation, because this interpretation never occurred to me. :-) It is technically true that the operations that have "acquire" or "release" in their title come in pairs, but the same can be said of things like `IRP_MJ_CREATE`/`IRP_MJ_CLEANUP`, `IRP_MJ_PREPARE_MDL_WRITE`/`IRP_MJ_MDL_WRITE_COMPLETE` and maybe others. From a filter's perspective, just because certain operations a "paired" doesn't mean your filter will last long enough to see both operations.

Granted, the "acquire"/"release" operations normally happen closer together than e.g. create/cleanup, but the principle is the same. Given that, I'd reword this paragraph to be more of a reminder and clarification:

```suggestion
Note that the terms "acquire" or "release" in some operation names, such as `IRP_MJ_ACQUIRE_FOR_CC_FLUSH` and `IRP_MJ_RELEASE_FOR_CC_FLUSH`, do not imply any guarantees that a minifilter will always see both operations. Just as with operations such as `IRP_MJ_CREATE` and `IRP_MJ_CLEANUP`, a minifilter may be prevented from observing the second operation by other factors, such as receiving an _[InstanceTeardownStartCallback](/windows-hardware/drivers/ddi/fltkernel/nc-fltkernel-pflt_instance_teardown_callback)_ before the second operation occurs.
```